### PR TITLE
Fix stack overflow when using non-existent MetaBot command

### PR DESCRIPTION
**Pull request by celwell**
Giving a Slack message to MetaBot with a command that doesn't exist (e.g., `metabot cool`) causes a stack overflow and responds with a null error message:
image

This PR fixes that bug, and another instance of the same bug within `metabot show`.

Also, an (unused?) let form was removed. Let me know if you want me to revert that (i.e., if it was being kept around for some reason). And, I have aggressive-indent-mode on, so let me know if you want me to undo the automatic indentation changes.

### CODE REVIEW

File: src/metabase/metabot/command.clj [hunk 1]
```diff
@@ -107,27 +107,19 @@
...
 (defn- list-cards []
   (filter-metabot-readable
-   (let [collection-ids                  (metabot-visible-collection-ids)
-         root-collection-perms?          (contains? collection-ids nil)
-         other-visible-collection-ids    (filter some? collection-ids)
-
-         root-collection-filter-clause   (when root-collection-perms?
-                                           [:= :collection_id nil])
-         other-collections-filter-clause (when (seq other-visible-collection-ids)
-                                           [:in :collection_id other-visible-collection-ids])]
```
Comment (camsaul) [inline src/metabase/metabot/command.clj]: I think I must have forgot to remove this when I put in the 

```clj
(collection/visible-collection-ids->honeysql-filter-clause (metabot-visible-collection-ids))
```

which accomplishes the same thing but more generally. good catch 👍

File: src/metabase/metabot/command.clj [hunk 2]
```diff
@@ -204,9 +196,9 @@
 
 (def ^:private kanye-quotes
   (delay
-    (log/debug (trs "Loading Kanye quotes..."))
-    (when-let [url (io/resource "kanye-quotes.edn")]
-      (edn/read-string (slurp url)))))
+   (log/debug (trs "Loading Kanye quotes..."))
```
Comment (camsaul) [inline src/metabase/metabot/command.clj]: [The body of `delay` forms should be indented by 2 spaces](https://guide.clojure.style/#body-indentation), please revert this indentation change

File: src/metabase/metabot/command.clj [hunk 3]
```diff
@@ -204,9 +196,9 @@
...
+   (when-let [url (io/resource "kanye-quotes.edn")]
+     (edn/read-string (slurp url)))))
```
Comment (camsaul) [inline src/metabase/metabot/command.clj]: [The body of `delay` forms should be indented by 2 spaces](https://guide.clojure.style/#body-indentation), please revert this indentation change

File: src/metabase/metabot/events.clj [hunk 4]
```diff
@@ -38,15 +38,15 @@
   (when (seq text)
     (second (re-matches #"^mea?ta?boa?t\s*(.*)$" text)))) ; handle typos like metaboat or meatbot
 
-(defn- respond-to-message! [message response]
```
Comment (camsaul) [inline src/metabase/metabot/events.clj]: 👍

**Review by camsaul — CHANGES_REQUESTED** [review]
This looks great.

One minor request: please revert that indentation change you made with the body of the `delay` form.

**Comment by camsaul** [comment]
Out of curiousity: if you're using `aggressive-indent-mode` are you using `clojure-mode` as well? Curious why it indented `delay` with one space for you.

Not sure how familiar you are with Emacs Lisp, but the way `clojure-mode` determines how to indent things is by looking a the value of `clojure-indent-function` in [the property list cell](https://www.gnu.org/software/emacs/manual/html_node/elisp/Symbol-Plists.html#Symbol-Plists) of the symbol with the same name. So for `delay` it does 

```emacs-lisp
;; Emacs Lisp
(get 'delay 'clojure-indent-function) ; -> 0
```

which tells it to indent it as a form with 0 args on the first line, in other words use 2 spaces e.g.

```clj
(delay
  ...)
```

(If you have CIDER connected to a REPL, it also looks at the `:style/indent` metadata for the var in question, but failing that it falls back to the value of `clojure-indent-function`).

**Comment by celwell** [comment]
Oh, apparently it was corrected in clojure-mode 5.11 (https://github.com/clojure-emacs/clojure-mode/issues/524). I've updated my package version, and `delay` now properly indents.

**Comment by camsaul** [comment]
Cool. Thanks!

**Pull request merged**